airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/AvroSerializedBuffer.java

```diff
@@ -52,6 +52,11 @@ protected void writeRecord(final AirbyteRecordMessage record) throws IOException
     dataFileWriter.append(avroRecordFactory.getAvroRecord(UUID.randomUUID(), record));
   }
 
+  @Override
+  protected void writeRecord(final String recordString, final long emittedAt) throws IOException {
+    // no op
```

Comment:
the AvroRecordFactory can also benefit from this. 

we don't need to also implement this here to keep things clean, however should left a todo for yourself here so this is not a dangling no op.